hystrix_client_test.go

```diff
@@ -320,6 +320,27 @@ func TestHystrixHTTPClientReturnsFallbackFailure(t *testing.T) {
 		commandConfig: hystrixCommandConfig,
 	})
 
+	_, err := client.Get("http://foobar.example", http.Header{})
+	assert.True(t, strings.Contains(err.Error(), "circuit open"))
+}
+
+func TestHystrixHTTPClientReturnsFallbackFailureWithAFallBackFunction(t *testing.T) {
+	hystrixCommandConfig := hystrix.CommandConfig{
+		Timeout:                10,
+		MaxConcurrentRequests:  100,
+		ErrorPercentThreshold:  10,
+		SleepWindow:            100,
+		RequestVolumeThreshold: 10,
+	}
+
+	client := NewHystrixHTTPClient(10, HystrixConfig{
+		commandName:   "some_command_name",
+		commandConfig: hystrixCommandConfig,
+		fallbackFunc: func(err error) error {
```

Comment:
Few things which can be added:

1. Add a test case for when the fallback function _doesn't_ return an error
2. Add an assertion to check whether the fallback function was called
3. Add assertion for the complete error text, not the substring

Reply:
https://github.com/afex/hystrix-go/blob/27fae8d30f1a3cbcab6618d2ee55a4cb43064376/hystrix/hystrix.go#L11
@sohamkamani it is not possible to pass in a function which doesn't return an error.

Reply:
the hystrix-go package doesn't implement an interface and therefore I cannot really mock the client to test the callability of the fallback function

Reply:
and, did the 3rd point. the assertion is now on the complete string.

Reply:
For (1), I meant we should test the err=nil case
For (2), you don't have to mock. Have a test variable `called := false`, set it to true inside the fallback, and then assert that `called==true`